src/apphosting/config.ts

```diff
@@ -172,6 +176,41 @@ export async function maybeAddSecretToYaml(secretName: string): Promise<void> {
   dynamicDispatch.store(path, projectYaml);
 }
 
+export async function exportConfig(
+  projectId: string,
+  cwd: string,
+  backendRoot: string,
+  environmentConfigFile?: string,
+) {
+  // Load apphosting.local.yaml file if it exists. Secrets should be added to the env list in this object and written back to the apphosting.local.yaml
+  let localAppHostingConfig: AppHostingYamlConfig = AppHostingYamlConfig.empty();
+
+  const localAppHostingConfigPath = resolve(backendRoot, APPHOSTING_LOCAL_YAML_FILE);
+  if (fs.fileExistsSync(localAppHostingConfigPath)) {
+    localAppHostingConfig = await AppHostingYamlConfig.loadFromFile(localAppHostingConfigPath);
+  }
+
+  const configToExport = await loadConfigToExport(cwd, environmentConfigFile);
+  const secretsToExport = configToExport.secrets;
+  if (!secretsToExport) {
+    logger.warn("No secrets found to export in the chosen App Hosting config files");
+    return;
+  }
+
+  const secretMaterial = await fetchSecrets(projectId, secretsToExport);
+  for (const [key, value] of secretMaterial) {
+    localAppHostingConfig.addEnvironmentVariable({
+      variable: key,
+      value: value,
+      availability: ["RUNTIME"],
+    });
+  }
+
+  // update apphosting.local.yaml
+  localAppHostingConfig.upsertFile(localAppHostingConfigPath);
+  logger.info(`Wrote secrets as environment variables to ${APPHOSTING_LOCAL_YAML_FILE}.`);
+}
+
```

Comment:
Nit: Mind cleaning up some of these check warnings while you are here? They seem pretty trivial to tidy up